BookNLPcharTracker.ipynb

```diff
@@ -2097,10 +40,61 @@
         "        }\n",
         "        self.booknlp = BookNLP(\"en\", self.model_params)\n",
         "\n",
+        "    def preprocess_text(self, input_file, output_file):\n",
+        "        \"\"\"\n",
+        "        Preprocess the book text by removing the introduction, table of contents, ending sections,\n",
+        "        text with leading spaces, and text enclosed in square brackets or between underscores.\n",
+        "        Removes unwanted punctuation but keeps sentence-ending punctuation marks.\n",
+        "        \"\"\"\n",
+        "        with open(input_file, 'r', encoding='utf-8') as f:\n",
+        "            text = f.read()\n",
+        "\n",
+        "        text = re.sub(r'_.*?_', '', text)\n",
```

Comment:
text = re.sub(r'_.*?_', '', text) I think we should delete this because it deletes some words from main text (some of them are pronouns)

Reply:
I change it to removing the sentence, not words